crates/turbo-tasks-memory/src/gc.rs

```diff
@@ -0,0 +1,199 @@
+use std::{
+    cmp::Reverse,
+    collections::HashMap,
+    time::{Duration, Instant},
+};
+
+use concurrent_queue::ConcurrentQueue;
+use nohash_hasher::BuildNoHashHasher;
+use turbo_tasks::{small_duration::SmallDuration, TaskId, TurboTasksBackendApi};
+
+use crate::{concurrent_priority_queue::ConcurrentPriorityQueue, MemoryBackend};
+
+/// The priority of a task for garbage collection.
+/// Any action will shrink the internal memory structures of the task in a
+/// transparent way.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+pub enum GcPriority {
+    // The order influences priority. Put the highest priority first.
```

Comment:
Nit: Could we do away with the `Reverse<GcPriority>` priority queueing if we just reversed the order of this enum?